From f6482e7b8c901f0f33363338e71cd7ae21be58f5 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Fri, 14 Jul 2017 09:59:12 +0800
Subject: [PATCH 2/5] irqchip: scfg-msi: Remove bad `__init' annotation

Fix:

WARNING: vmlinux.o(.data+0x57a70): Section mismatch in reference from the variable ls_scfg_msi_driver to the function .init.text:ls_scfg_msi_probe()
The variable ls_scfg_msi_driver references
the function __init ls_scfg_msi_probe()
If the reference is valid then annotate the
variable with __init* or __refdata (see linux/init.h) or name the variable:
*_template, *_timer, *_sht, *_ops, *_probe, *_probe_one, *_console

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/irqchip/irq-ls-scfg-msi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/irqchip/irq-ls-scfg-msi.c b/drivers/irqchip/irq-ls-scfg-msi.c
index eba6ecc..ae661b9 100644
--- a/drivers/irqchip/irq-ls-scfg-msi.c
+++ b/drivers/irqchip/irq-ls-scfg-msi.c
@@ -425,7 +425,7 @@ _err:
 	return err;
 }
 
-static int __init ls_scfg_msi_probe(struct platform_device *pdev)
+static int ls_scfg_msi_probe(struct platform_device *pdev)
 {
 	struct ls_scfg_msi *msi_data;
 	const struct soc_device_attribute *match;
-- 
2.9.3

